pkg/workflow/controllers/parallel_node_reconciler.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+)
+
+// ParallelNodeReconciler watches on nodes which type is Parallel
+type ParallelNodeReconciler struct {
+	*ChildrenNodesFetcher
+	kubeClient    client.Client
+	eventRecorder record.EventRecorder
+	logger        logr.Logger
+}
+
+func NewParallelNodeReconciler(kubeClient client.Client, eventRecorder record.EventRecorder, logger logr.Logger) *ParallelNodeReconciler {
+	return &ParallelNodeReconciler{
+		ChildrenNodesFetcher: NewChildrenNodesFetcher(kubeClient, logger),
+		kubeClient:           kubeClient,
+		eventRecorder:        eventRecorder,
+		logger:               logger,
+	}
+}
+
+// Reconcile is extremely like the one in SerialNodeReconciler, only allows the parallel schedule, and respawn **all** the children tasks during retry
+func (it *ParallelNodeReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	startTime := time.Now()
+	defer func() {
+		it.logger.V(4).Info("Finished syncing for parallel node",
+			"node", request.NamespacedName,
+			"duration", time.Since(startTime),
+		)
+	}()
+
+	ctx := context.TODO()
+
+	node := v1alpha1.WorkflowNode{}
+	err := it.kubeClient.Get(ctx, request.NamespacedName, &node)
+	if err != nil {
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// only resolve parallel nodes
+	if node.Spec.Type != v1alpha1.TypeParallel {
+		return reconcile.Result{}, nil
+	}
+
+	it.logger.V(4).Info("resolve parallel node", "node", request)
+
+	// make effects, create/remove children nodes
+	err = it.syncChildrenNodes(ctx, node)
```

Comment:
umm shouldn't it be `ChildNodes` rather than `ChildrenNodes`?

Reply:
fixed, PTAL